### DISCUSSION TITLE
Strange dashboard layout, Change Log widget missing Object Change model

### DISCUSSION THREAD

**Discussion by ghaberek** [netbox-community/netbox#12535] [Help Wanted!] [answered]
I was excited to try the new dashboard feature, only to discover that mine is not working correctly. I have a feeling that something may have gone wrong during the upgrade but I'm not sure how to track it down. Here's what I know:

- The dashboard area is overflowing the page. You can see the paging buttons for the Change Log widget are cut off at the bottom of the page and I cannot scroll down far enough to see the "Unlock Dashboard" button.
- When I edit the Change Log widget, the configuration is missing the "Extras > Object Change" model, which is the relevant item for that widget. Instead it has "Circuits > Circuit" selected, which is the first item in the list.
- I have the [Topology Views](https://github.com/mattieserver/netbox-topology-views) plugin installed and up to date (v3.5.0)
- I _did_ have the [Device Map](https://github.com/drygdryg/netbox-plugin-device-map) plugin installed, and this was causing errors during the upgrade. I thought I had worked these out by using [my own fork](https://github.com/ghaberek/netbox-plugin-device-map) to correct (what I thought were) a few [simple changes](https://github.com/ghaberek/netbox-plugin-device-map/commit/322c1e7cdde271d3e3bbb6837c2d492742ba289c) to the imported module names.
- I have since removed the Device Map plugin and re-run the upgrade script, but the dashboard is still messed up.
- Using the `/extras/dashboard/reset/` option did not help.
- I upgraded to 3.5.1 this morning, hoping it might have had a fix for this, but unfortunately that did not help.

Did I do something to muck up the database schema? How can I correct this?

### My site

|![image](https://github.com/netbox-community/netbox/assets/12944771/ab273225-1766-42d5-828c-5c18a308928c)|![image](https://github.com/netbox-community/netbox/assets/12944771/d435c2e4-baec-4f41-9fc2-ddc0b495b4c4)|
|:---:|:---:|
| Dashboard | Change Log |

### Demo site

|![image](https://github.com/netbox-community/netbox/assets/12944771/ba6117ee-1660-41ef-a16f-74873de4e91c)|![image](https://github.com/netbox-community/netbox/assets/12944771/904af911-3049-49b6-b273-7812e1c1c287)|
|:---:|:---:|
| Dashboard | Change Log |

**Comment by kkthxbye-code** [comment]
> When I edit the Change Log widget, the configuration is missing the "Extras > Object Change" model, which is the relevant item for that widget. Instead it has "Circuits > Circuit" selected, which is the first item in the list.

This was fixed in 3.5.1, can you please verify that you are seeing it in 3.5.0?

Not sure about the rendering stuff, resetting the dashboard should clear the settings. Do you have a DEFAULT_DASHBOARD configuration set? You could also try to create a new user and login as that to make sure there's nothing in the user config messing things up.

**Reply by ziggekatten** [reply]
Don't forget to clear browser cache after upgrade!

Weird things I had with the dashboard was related to this. Try sn icognito window to see if thats the case.

**Comment by ghaberek** [comment, marked as answer]
Soooooo... I assumed I was on 3.5.1 but, since the bottom of the page was cut off, I didn't actually _see_ the version. Navigating to a another page clearly showed 3.5.0 so mea culpa on that.

I'm trying to backtrack and figure out how I got here and hopefully prevent myself from doing this again. For a while I was downloading release archives and extracting them but I found the whole process tedious so I switched to using git several releases ago.

Turns out when you follow the [upgrade directions](https://docs.netbox.dev/en/stable/installation/upgrading/#option-b-clone-the-git-repository) and `git checkout master` reports _"Your branch is up to date with 'origin/master'"_ that doesn't mean "there are no changes to pull" and you still need to run `git pull origin master` to get the changes. With as much as I use git elsewhere, I guess I'm not as well versed in being the _consumer_ of a repo as _opposed_ to the author.

What's the best way to determine my version of NetBox from the local directory?

Using `git branch` always shows _master_ and `git tag` only showed _v3.4.1_ (I'm guessing that's when I switched to using git). I had to use a plain `git pull` to get all the tags since v3.4.1. I suppose something like `git log -n5` would suffice. There I can see the last _"Release vX.Y.X"_ in the commit notes.

Thanks for the help. New dashboard works great! 👍
